### PR TITLE
feat: I/O safety for 'sys/memfd' & 'sys/event' & 'sys/eventfd'

### PR DESCRIPTION
**Pull request by SteveLauC**
#### What this PR does:

Adds I/O safety for moduels:
1. `sys/memfd`
2. `sys/event`
3. `sys/eventfd`

-----

BYW, I called `rustfmt` on these 4 files, which introduces some noise, sorry about this.

### CODE REVIEW

File: src/sys/event.rs [hunk 1]
```diff
@@ -267,8 +267,8 @@ impl KEvent {
     }
 }
 
-pub fn kevent(
-    kq: RawFd,
+pub fn kevent<Fd: AsFd>(
+    kq: &Fd,
```
Comment (JonathanWoollett-Light) [inline src/sys/event.rs:271]: It is better to pass `Fd` as not a reference, consider [`impl<T: AsFd> AsFd for &T`](https://doc.rust-lang.org/nightly/std/os/fd/trait.AsFd.html#impl-AsFd-for-%26T).

Comment (SteveLauC) [reply src/sys/event.rs:271]:
```rust
use std::{fs::File, os::unix::io::AsFd};

fn syscall<Fd: AsFd>(_fd: Fd) {}

fn main() {
    let f1: File = File::open(".").unwrap();
    let f2: File = f1.try_clone().unwrap();

    syscall(f1);  // We can pass T: AsFd
    syscall(&f2); // and ref to T
}
```
I get it, though all my PRs use `Fd: AsFd(fd: &Fd)` in the current implementations...

cc @asomers, any thoughts on this, which type should we use?

Comment (SUPERCILEX) [reply src/sys/event.rs:271]: Agreed with @JonathanWoollett-Light, see my comment here: https://github.com/nix-rust/nix/issues/1750#issuecomment-1343124228

File: src/sys/epoll.rs [hunk 2]
```diff
@@ -78,22 +78,22 @@ impl EpollEvent {
...
 /// const MILLIS: u64 = 100;
-/// 
+///
 /// // Create epoll
 /// let epoll = Epoll::new(EpollCreateFlags::empty())?;
-/// 
+///
 /// // Create eventfd & Add event
-/// let eventfd = unsafe { OwnedFd::from_raw_fd(eventfd(0, EfdFlags::empty())?) };
+/// let eventfd = unsafe { eventfd(0, EfdFlags::empty())? };
```
Comment (asomers) [inline src/sys/epoll.rs:86]: Why does this need an `unsafe` block?

**Review by asomers — CHANGES_REQUESTED** [review]
Could you please split the formatting changes out into a separate commit?

**Comment by SUPERCILEX** [comment]
Made the formatting changes here: https://github.com/nix-rust/nix/pull/1935. CI checks are clearly still broken, I'll try and figure out why if I have time.

File: src/sys/epoll.rs (hunk 2 above)
Comment (SteveLauC) [reply src/sys/epoll.rs:86]: Missed that, sorry about this.

**Comment by SteveLauC** [comment]
> Could you please split the formatting changes out into a separate commit?

Let't merge #1935 first, then I will rebase my branch.

**Comment by asomers** [comment]
I think you need to rebase to make the formatting changes go away.  And the OSX aarch64 failure is spurious.

**Comment by SteveLauC** [comment]
> I think you need to rebase to make the formatting changes go away. And the OSX aarch64 failure is spurious.

Rebased.

**Review by asomers — APPROVED** [review]
bors r+

**Pull request merged**